src/ophys_etl/pipelines/dff.py

```diff
@@ -0,0 +1,166 @@
+                     "`output_file` containing ROI IDs associated with "
+                     "traces.")
+        )
+    output_dataset = fields.Str(
+        required=False,
+        default="data",
+        description=("h5 dataset key used to store the computed dff traces "
+                     "in `output_file`.")
+    )
+    sigma_dataset = fields.Str(
+        required=False,
+        default="sigma_dff",
+        description=("h5 dataset key used to store the estimated noise "
+                     "standard deviation for the dff traces in `output_file`.")
+    )
+    baseline_frames_dataset = fields.Str(
+        required=False,
+        default="num_small_baseline_frames",
+        description=("h5 dataset key used to store the number of small "
+                     "baseline frames (where the computed baseline of the "
+                     "fluorescence trace was smaller than its estimated "
+                     "noise standard deviation) in `output_file`.")
+    )
+    long_baseline_filter_s = fields.Int(
+        required=False,
+        default=600,
+        description=("Number of seconds to use in the rolling median "
+                     "filter for for computing the baseline activity. "
+                     "The length of the filter is the frame rate of the "
+                     "signal in Hz * the long baseline filter seconds ("
+                     "+1 if the result is even, since the median filter "
+                     "length must be odd).")
+    )
+    short_filter_s = fields.Float(
+        required=False,
+        default=3.333,
+        description=("Number of seconds to use in the rolling median "
+                     "filter for the short timescale detrending. "
+                     "The length of the filter is the frame rate of the "
+                     "signal in Hz * the short baseline filter seconds ("
+                     "+1 if the result is even, since the median filter "
+                     "length must be odd).")
+    )
+
+    @post_load
+    def filter_s_to_frames(self, item, **kwargs):
+        """Convert number of seconds to number of frames for the
+        filters `short_filter_s`, `long_baseline_filter_s`. If the
+        number of frames is even, add 1."""
+        short_frames = int(np.round(
+            item["movie_frame_rate_hz"] * item["short_filter_s"]))
+        long_frames = int(np.round(
+            item["movie_frame_rate_hz"] * item["long_baseline_filter_s"]))
+        # Has to be odd
+        item["short_filter_frames"] = (
+            short_frames if short_frames % 2 else short_frames + 1)
+        item["long_filter_frames"] = (
+            long_frames if long_frames % 2 else long_frames + 1)
+        return item
+
+
+class DffJob(ArgSchemaParser):
+    """
+    This is the job runner for the dF/F computation from F (fluorescence)
+    traces. The primary data input is the h5 file produced by neuropil
+    subtraction, which contains the "corrected fluorescence trace"
```

Comment:
this name "corrected" as input is confusing. According to your doc, the input is "cell" fluorescence trace, and we are correcting it in this module.

Reply:
or maybe this is the "intermediate" format. So, this is actually getting neuropil-corrected trace as input?

Reply:
I'm not sure... I think "corrected fluorescence trace" actually means something. It's in the SDK all over.

Reply:
This is the "intermediate format" in my doc, yes.

Reply:
can you call it neuropil-corrected trace?

Reply:
sorry, just read your above comment. Ok. If it's true that "corrected" and "neuropil-corrected" are truly different, this is the exact ambiguity it would be good to eliminate. @ledochowitsch is there a difference?

Reply:
discussed with Marina that neuropil-corrected fluorescence trace is good